config/v1/types_infrastructure.go

```diff
@@ -675,6 +738,19 @@ type GCPPlatformStatus struct {
 	// +optional
 	// +nullable
 	CloudLoadBalancerConfig *CloudLoadBalancerConfig `json:"cloudLoadBalancerConfig,omitempty"`
+
+	// serviceEndpoints specifies endpoints that override the default endpoints
+	// used when creating clients to interact with GCP services.
+	// When not specified, the default endpoint for the GCP region will be used.
+	// Only 1 endpoint override is permitted for each GCP service.
+	// The maximum number of endpoint overrides allowed is 9.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=9
+	// +kubebuilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x.name == y.name))",message="only 1 endpoint override is permitted per GCP service name"
+	// +optional
+	// +openshift:enable:FeatureGate=GCPCustomAPIEndpoints
+	ServiceEndpoints []GCPServiceEndpoint `json:"serviceEndpoints,omitempty"`
```

Comment:
Should be in status and not spec, consumers must read from status, installer will populate the status like it does for other platforms